File: ios/Classes/Model/CardFactory.swift
//
//  CardFactory.swift
//  com.twocheckout.twocheckout_flutter
//
//  Copyright © 2023 DevCrew I/O
//

import Foundation

final class CardFactory {
    private init() {}

    static func createCardFromMap(_ arguments: [String: Any]) -> Card {
        return Card(name: arguments["name"] as? String ?? "",
                    creditCard: arguments["creditCard"] as? String ?? "",
                    cvv: arguments["cvv"] as? String ?? "",
                    expirationDate: arguments["expirationDate"] as? String ?? "",
                    scope: arguments["scope"] as? String)
    }
}

File: android/src/main/kotlin/com/twocheckout/twocheckout_flutter/Model/CardFactory.kt
//
//  CardFactory.kt
//  com.twocheckout.twocheckout_flutter
//
//  Copyright © 2023 DevCrew I/O
//

package com.twocheckout.twocheckout_flutter.Model
import com.twocheckout.twocheckout_flutter.datapack.CreditCardInputResult
import com.twocheckout.twocheckout_flutter.datapack.PayerCardData

object CardFactory {
    fun createCardFromMap(arguments: Map<String, Any>): CreditCardInputResult {
        val cardInputObject = CreditCardInputResult()
        cardInputObject.name = arguments["name"] as? String ?: ""
        val cardPayData = PayerCardData()
        cardPayData.expirationDate = arguments["expirationDate"] as? String ?: ""
        cardPayData.creditCard = arguments["creditCard"] as? String ?: ""
        cardPayData.cvv = arguments["cvv"] as? String ?: ""
        cardInputObject.cardData = cardPayData
        return cardInputObject
    }
}
